java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -26,14 +28,16 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Properties;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class WorkspaceDriver {
 
-  private static Path tmp;
+  static Path tmp;
   private static Map<String, Path> bazelVersions;
   private static Path runfileDirectory = Paths.get(System.getenv("TEST_SRCDIR"));
+  static Properties properties;
```

Comment:
why does this need to be public?
we have global mutable state here :(
can we make it private and expose a method to query specific properties?
maybe even make it non static to ease our own tests (not sure about this last bullet)

Reply:
So the reason I did that is that, since it's package-private, it's an implementation detail, not something that is exposed to the user.  I agree this is ugly, but my reasoning was that the best way to deal with that was, instead of having a bunch of static objects, to have a companion class/singleton where to put the config, but I thought this refactoring would be better done, in full, in another PR.